File: platform/sdk/nodejs/examples/cloud/cloud-024_worm_compliance_immutable.js
const { CloudClient } = require("../../xuanji-sdk-cloud");
const client = new CloudClient({ region: "cn-east-1" });
client.createBucket("comp-bucket");
client.putObject("comp-bucket", "imm.txt", "immutable content");
client.wormRetention1y("comp-bucket", "imm.txt");
const result = client.wormComplianceImmutable("comp-bucket", "imm.txt");
if (!result.immutable || result.canDelete) process.exit(1);
console.log("XJ-OK: cloud-024_worm_compliance_immutable");
process.exit(0);

File: platform/sdk/python/examples/cloud/cloud-024_worm_compliance_immutable.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from mox_sdk_cloud import CloudClient

client = CloudClient({"region": "cn-east-1"})
client.createBucket("comp-bucket")
client.putObject("comp-bucket", "imm.txt", "immutable content")
client.wormRetention1y("comp-bucket", "imm.txt")
result = client.wormComplianceImmutable("comp-bucket", "imm.txt")
if not result["immutable"] or result["canDelete"]:
    sys.exit(1)
print("XJ-OK: cloud-024_worm_compliance_immutable")
sys.exit(0)
